project/threads/models.py

```diff
@@ -208,6 +217,15 @@ def thread_sorted_by_score(self, civis_queryset, requested_user_id):
             queryset.all(), key=lambda c: c.score(requested_user_id), reverse=True
         )
 
+    def problems(self, thread_id):
+        return self.filter(thread__id=thread_id, c_type="problem")
+
+    def causes(self, thread_id):
+        return self.filter(thread__id=thread_id, c_type="cause")
+
+    def solutions(self, thread_id):
+        return self.filter(thread__id=thread_id, c_type="solution")
```

Comment:
How will these methods be accessed? E.g. will it be something like `Civi.objects.filter(id=<some-id).problems()`? 

- https://docs.djangoproject.com/en/4.1/topics/db/managers/#adding-extra-manager-methods

Ideally, we would have direct properties on the Civi model instance, such as from a given civi: 
- `civi.problems`
- `civi.causes`
- `civi.solutions`

Reply:
After further reading this PR, let's add these properties directly to the `Thread` model.

- `thread.problems`
- `thread.causes`
- `thread.solutions`

Reply:
```py
class Thread(models.Model):
    ...
    @property
    def problems(self):
        return self.civis.filter(c_type="problem")

    @property
    def causes(self):
        return self.civis.filter(c_type="cause")

    @property
    def solutions(self):
        return self.civis.filter(c_type="solution")
```

Reply:
`Civi.objects.filter(id=<some-id).problems()` is not what we want. A civi can be either a **problem** or a **cause** or a **solution**. (Please see `c_type` field.) `Civi.objects.filter(id=<some-id)` will return only one civi. A civi doesn't have problems. Instead, a civi itself can be a problem. So `Civi.objects.filter(id=<some-id)` might return one civi that is a problem civi.

What we want is to return **problem** civis or **cause** civis or **solution** civis of a thread. We can describe it as:
`Civi.objects.filter(thread__in=thread_in).problems()`. As you pointed out, it is also useful: `Thread.objects.get(id=thread_in).problems()`.  I will add these properties to the `Thread` model but not `Thread Manager` because we are interested in problems of one thread not multiple threads. On the other hand, we want to get multiple civis which are problems so we should add these properties to `CiviManager` at this time.

Reply:
So these are not valid:

- civi.problems
- civi.causes
- civi.solutions

let me correct them 
- civi.c_type = problem
- civi.c_type = cause
- civi.c_type = solution

Instead of getting problems of a civi we should get problem civis:
- civis.problems
- civis.causes
- civis.solutions

preferable:

- thread.civis.problems
- thread.civis.causes
- thread.civis.solutions

Let's take one more step as you mentioned:

- thread.problems
- thread.causes
- thread.solutions

Reply:
> On the other hand, we want to get multiple civis which are problems so we should add these properties to CiviManager at this time.

I'm not aware of a current use case for the CiviManager methods under discussion.

I think we're in agreement to move the ModelManager methods to the Thread model as properties since those are immediately useful.